google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -172,43 +172,73 @@ private void invokePanelValidationUpdate(Runnable runnable) {
   }
 
   @VisibleForTesting
-  public TextFieldWithBrowseButton getCloudSdkDirectoryField() {
+  TextFieldWithBrowseButton getCloudSdkDirectoryField() {
     return cloudSdkDirectoryField;
   }
 
   @VisibleForTesting
-  JTextPane getWarningMessage() {
-    return warningMessage;
+  JRadioButton getManagedRadioButton() {
+    return managedRadioButton;
   }
 
-  public boolean isModified() {
-    CloudSdkService sdkService = CloudSdkService.getInstance();
+  @VisibleForTesting
+  JCheckBox getEnableAutomaticUpdatesCheckbox() {
+    return enableAutomaticUpdatesCheckbox;
+  }
 
-    String cloudSdkDirectoryFieldValue =
-        getCloudSdkDirectoryText() != null ? getCloudSdkDirectoryText() : "";
-    String currentCloudSdkDirectory =
-        sdkService.getSdkHomePath() != null ? sdkService.getSdkHomePath().toString() : "";
-    return !cloudSdkDirectoryFieldValue.equals(currentCloudSdkDirectory);
+  @VisibleForTesting
+  JRadioButton getCustomRadioButton() {
+    return customRadioButton;
+  }
+
+  public boolean isModified() {
+    return settingsModified;
   }
 
   public void apply() throws ConfigurationException {
-    if (CloudSdkValidator.getInstance()
-        .validateCloudSdk(getCloudSdkDirectoryText())
-        .contains(CloudSdkValidationResult.MALFORMED_PATH)) {
-      throw new ConfigurationException(
-          GctBundle.message("appengine.cloudsdk.location.badchars.message"));
+    CloudSdkServiceUserSettings sdkServiceUserSettings = CloudSdkServiceUserSettings.getInstance();
+
+    if (customRadioButton.isSelected()) {
+      String customSdkPathText = getCloudSdkDirectoryText();
+      if (CloudSdkValidator.getInstance()
+          .validateCloudSdk(customSdkPathText)
+          .contains(CloudSdkValidationResult.MALFORMED_PATH)) {
+        throw new ConfigurationException(
+            GctBundle.message("appengine.cloudsdk.location.badchars.message"));
+      }
+
+      sdkServiceUserSettings.setCustomSdkPath(customSdkPathText);
+      CloudSdkService.getInstance().setSdkHomePath(customSdkPathText);
     }
 
-    CloudSdkService.getInstance().setSdkHomePath(getCloudSdkDirectoryText());
+    sdkServiceUserSettings.setUserSelectedSdkServiceType(selectedCloudSdkServiceType);
+
+    sdkServiceUserSettings.setEnableAutomaticUpdates(enableAutomaticUpdatesCheckbox.isSelected());
+
+    // settings are applied and saved, clear modification status
+    settingsModified = false;
   }
 
   public void reset() {
-    CloudSdkService sdkService = CloudSdkService.getInstance();
+    CloudSdkServiceUserSettings sdkServiceUserSettings = CloudSdkServiceUserSettings.getInstance();
+
+    CloudSdkServiceUserSettings.CloudSdkServiceType selectedSdkServiceType =
+        sdkServiceUserSettings.getUserSelectedSdkServiceType();
+    switch (selectedSdkServiceType) {
+      case MANAGED_SDK:
+        managedRadioButton.doClick();
```

Comment:
why doClick instead of setSelected?

Reply:
I use `ActionListener` to listen for their actions, but `ActionListener`s are not called after `setSelected` - I think `ItemListener` can be used instead but they are a bit more verbose to use. Does it feel slightly confusing? We can use selection listeners and methods instead.

Reply:
just curios why you need the listeners to be called from the reset method? I see that later you are defaulting the other properties (like auto update).

Reply:
That's sort of unified way to set up UI state properly - `reset()` is called when panel is opened, and UI state is reset based on settings - selecting radio button brings the state of components into desired state (i.e. enables/disables based on chosen SDK).

Reply:
Ah yeah. There are other things like enable / disable controls which happen via the listener. Makes sense. 

I’ll leave the doClick up to you since I don’t feel too strongly either way. Personally I tend to just use the other kind of listener that gets fired on setSelected.

Reply:
I tried `setSelected` but forgot it only fires events when state is actually changed - i.e. false -> true or vice versa. In this case we want the action to be called for sure, so `doClick()` works better.